docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
         // First request to initialize the type server.
         Initialize = 'typeServer/initialize',
         // Request from client to shut down the type server.
-        Shutdown = 'typeServer/shutdown',
+        ShutDown = 'typeServer/shutdown',
         // Request from client to get the current snapshot of the type server.
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
         // Request to get diagnostics for a specific file.
         GetDiagnostics = 'typeServer/getDiagnostics',
         // Request to get the version of diagnostics for a specific file.
         GetDiagnosticsVersion = 'typeServer/getDiagnosticsVersion',
         // Request to get the type information for a specific node.
         GetType = 'typeServer/getType',
-        // Request to get the union types for a specific type.
+        // Request to get the collection of subtypes that make up a union type.
         GetUnionTypes = 'typeServer/getUnionTypes',
-        // Request to find a member of a type.
-        GetTypeMember = 'typeServer/getTypeMember',
-        // Request to get all overloads of a function or method.
+        // Request to find an attribute of a class.
+        GetTypeAttribute = 'typeServer/getTypeAttribute',
+        // Request to get all overloads of a function or method. The returned value doesn't include the implementation signature.
         GetOverloads = 'typeServer/getOverloads',
-        // Request to get the effective meta class of a type.
-        GetEffectiveMetaClass = 'typeServer/getEffectiveMetaClass',
+        // Request to get the meta class of a type.
+        GetMetaclass = 'typeServer/getMetaclass',
         // Request to get the type of a declaration.
         GetTypeOfDeclaration = 'typeServer/getTypeOfDeclaration',
```

Comment:
seems like we can avoid this API by using the `Node` from the `Declaration`. I think the only case it won't work is for `Intrinsic`.

Reply:
Pyright's [implementation](https://github.com/microsoft/pyright/blob/5021af8f0545386b99e916b764bb8968d362bd7f/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L22197) of this is more involved than the same thing for a node. It could certainly be implemented internally using the node but a declaration seems like it can get more information.